### ISSUE TITLE
Repeated Events in FileMonitor

### ISSUE THREAD

**Issue by tobyweston** [pathikrit/better-files#140]
I've encountered this as part of #137 but in the `process` method, it handles "repeated" events;

     repeat(event.count())(onEvent(event.kind(), target))

Given the docs

> Returns the event count. If the event count is greater than {@code 1} then this is a repeated event.

and the gist of [what-does-it-mean-for-an-event-to-be-repeated?](http://stackoverflow.com/questions/21998659/what-does-it-mean-for-an-event-to-be-repeated?noredirect=1&lq=1) talks about, I can't think why we want to fire multiple events like this.

Basically, I'm suggesting the following and wanted to make sure I'm not missing something.

     onEvent(event.kind(), target)

What do you think?

**Comment by pathikrit** [comment]
or even better, let the client handle it - will change signature to be `onEvent(event.kind(), target, event.count())`

**Comment by tobyweston** [comment]
Might be nice to offer both, have `override def onCreate(file: File, count: Int) = {}` default to calling the `onCreate` with an argument of `1`? Or use a default value of `1`? 

    override def onCreate(file: File, count: Int, count: Int = 1) = ...

I'm thinking to avoid an API breaking change
